server/service/src/processors/general_processor.rs

```diff
@@ -0,0 +1,145 @@
+use repository::{
+    ChangelogFilter, ChangelogRepository, ChangelogRow, ChangelogTableName, EqualFilter, KeyType,
+    RepositoryError, StorageConnection, TransactionError,
+};
+use thiserror::Error;
+
+use crate::{
+    cursor_controller::CursorController,
+    email::EmailServiceError,
+    processors::log_system_error,
+    service_provider::{ServiceContext, ServiceProvider},
+};
+
+use super::contact_form::QueueContactEmailProcessor;
+
+#[derive(Error, Debug)]
+pub(crate) enum ProcessorError {
+    #[error("{0:?} not found: {1:?}")]
+    RecordNotFound(String, String),
+    #[error("{0:?}")]
+    DatabaseError(RepositoryError),
+    #[error("{0:?}")]
+    EmailServiceError(EmailServiceError),
+}
+
+const CHANGELOG_BATCH_SIZE: u32 = 20;
+
+#[derive(Clone)]
+pub enum ProcessorType {
+    ContactFormEmail,
+}
+
+impl ProcessorType {
+    pub(super) fn get_processor(&self) -> Box<dyn Processor> {
+        match self {
+            ProcessorType::ContactFormEmail => Box::new(QueueContactEmailProcessor),
```

Comment:
Only thought is whether this should support a vec of processors, like we have for invoices etc... there might end up being multiple things we want to process with the contact form? But guess we can add that when we get there 😁